model/common/src/icon4py/model/common/grid/base.py

```diff
@@ -183,6 +183,14 @@ def _construct_connectivity(self, dim, from_dim, to_dim):
             )
         else:
             neighbor_table = self._neighbor_tables[dim]
+
+        if (dimension_size := self.size[to_dim]) <= neighbor_table.shape[0]:
+            # In Fortran, all connectivities have `nproma` size.
+            # Here we are restricting the connectivity to the actual size as this is currently used
+            # for sizing GT4Py temporaries.
+            _log.info(f"Restricting connectivity for {dim} to size of {to_dim} ({dimension_size}).")
+            neighbor_table = neighbor_table[:dimension_size, :]
```

Comment:
For me is fine to do it unconditionally, the if is never reached in greenline. What I mostly dislike is the comment, because I really don't want to see `nproma` mentioned anywhere in icon4py... :-/